src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -404,6 +415,27 @@ private void DiscoverHubMethods()
             }
         }
 
+        private async Task<bool> IsHubMethodAuthorized(ClaimsPrincipal principal, IEnumerable<IAuthorizeData> policies)
+        {
+            if (policies != null)
+            {
+                using (var scope = _serviceScopeFactory.CreateScope())
```

Comment:
Because thats how you use `IServiceScopeFactory` which is all we have access to regarding services